docs/spec.md

```diff
@@ -3020,80 +3027,123 @@ behavior is undefined. More formally, for all `i1 < i2` from `indices(result)`,
-| (I4)  | `collapsed_slice_dims` | 1-dimensional tensor constant of type `si64` | (C1), (C6-C8), (C13)         |
-| (I5)  | `start_index_map`      | 1-dimensional tensor constant of type `si64` | (C3), (C9), (C10)            |
-| (I6)  | `index_vector_dim`     | constant of type `si64`                      | (C2), (C3), (C13)            |
-| (I7)  | `slice_sizes`          | 1-dimensional tensor constant of type `si64` | (C8), (C11-C13)              |
-| (I8)  | `indices_are_sorted`   | constant of type `i1`                        |                              |
+| Label | Name                          | Type                                         | Constraints                                |
+|-------|-------------------------------|----------------------------------------------|--------------------------------------------|
+| (I1)  | `operand`                     | tensor or per-tensor quantized tensor        | (C1), (C8), (C11), (C17), (C19-C21), (C23) |
+| (I2)  | `start_indices`               | tensor of integer type                       | (C2-C3), (C14), (C17), (C22)               |
+| (I3)  | `offset_dims`                 | 1-dimensional tensor constant of type `si64` | (C1), (C4-C5), (C22)                       |
+| (I4)  | `collapsed_slice_dims`        | 1-dimensional tensor constant of type `si64` | (C1), (C6-C9), (C22)                       |
+| (I5)  | `operand_batching_dims`       | 1-dimensional tensor constant of type `si64` | (C1), (C6), (C10-C12), (C16-C18), (C22)    |
+| (I6)  | `start_indices_batching_dims` | 1-dimensional tensor constant of type `si64` | (C13-C17)                                  |
+| (I7)  | `start_index_map`             | 1-dimensional tensor constant of type `si64` | (C3), (C18-C19)                            |
+| (I8)  | `index_vector_dim`            | constant of type `si64`                      | (C2-C3), (C15), (C22)                      |
+| (I9)  | `slice_sizes`                 | 1-dimensional tensor constant of type `si64` | (C9), (C12), (C20-C22)                     |
+| (I10) | `indices_are_sorted`          | constant of type `i1`                        |                                            |
 
 #### Outputs
 
 | Name     | Type                                  | Constraints     |
 |----------|---------------------------------------|-----------------|
-| `result` | tensor or per-tensor quantized tensor | (C5), (C13-C14) |
+| `result` | tensor or per-tensor quantized tensor | (C5), (C22-C23) |
 
 #### Constraints
 
-* (C1) `rank(operand) = size(offset_dims) + size(collapsed_slice_dims)`.
+* (C1) `rank(operand) = size(offset_dims) + size(collapsed_slice_dims) +
+       size(operand_batching_dims)`.
 * (C2) `0 <= index_vector_dim <= rank(start_indices)`.
 * (C3) `size(start_index_map) =
        index_vector_dim < rank(start_indices) ?
        dim(start_indices, index_vector_dim) : 1`.
 * (C4) `is_unique(offset_dims) and is_sorted(offset_dims)`.
```

Comment:
I only splitted it for cases where the is_unique check is done on multiple vectors, e.g. collapsed_slice_dims and operand_batch_dims. cases like offset_dims can stay as is since you still only care if they are unique locally. wdyt?

I can see why it would make sense to just always have is_unique and is_sorted separate, but can we do this in a follow up as this is quite a big change and it would be safer to make it as a single change?